switching/input/messages/R1.py

```diff
@@ -732,3 +748,124 @@ def detall_resultat(self):
             pass
         return ref
 
+
+class MinimumFieldsChecker(object):
```

Comment:
If an instance of this object has not attribures We can do two things:

- Covert all of the methods `check_XXX` to [staticmethod](https://docs.python.org/2/library/functions.html#staticmethod) and then call checkers with `MiniumFieldCheck.check_XXX`
- Create a MiniumFieldChecker with the `R1` object as attribute in the `__init__`
  - In this way I think performs better because in all the `check_XXX` methods is not necessary to push in the method stach the `r1` object as is in the self instance.
  - Another improvement is that we can to something similar to (pseudocode):

```python
checker = MinimumFieldsChecker(r1)
checker.check()
```
   And the `check()` method is responsible to do the part of getting the `subtype` and the `min_fields` and run all the checks for this `subtype`.

```python
class MinimumFieldsChecker(object):
    def __init__(self, r1):
        self.r1 = r1

    def check(self):
        errors = []
        for field in self.r1.get_minimum_fields():
            valid = getattr(self, 'check_{0}'.format(field), None)
            if not valid:
                errors.append(valid)
        return errors
```